### PR TITLE
Add `mc-sgx-alloc` crate

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Adding template files for the `mc-sgx-alloc` crate

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/5" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/5?utm_source=stack-comment).

File: .gitignore [hunk 1]
```diff
@@ -2,9 +2,5 @@
 # will have compiled files and executables
 /target/
 
-# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
-# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
-Cargo.lock
-
```
Comment (jcape) [inline .gitignore:8]: I'm leaning towards wanting to leave this in, and just use the full "x.y.x" versions for dependencies, to trigger dependabot.

**Review by jcape — APPROVED** [review]
LGTM, we should chat about how we want to use `Cargo.lock` in libraries going forward...

File: .gitignore (hunk 1 above)
Comment (nick-mobilecoin) [reply .gitignore:8]: :facepalm: I read that backwards, i'll add back

**Review by NotGyro — APPROVED** [review]
Looks good to me - initial commit but docs and project structure look good

**Pull request merged**